src/target/riscv/riscv-013.c

```diff
@@ -3647,20 +3691,18 @@ static int read_memory_progbuf_inner_startup(struct target *target,
 		goto clear_abstractauto_and_fail;
 
 	uint32_t abstractcs;
-
 	if (wait_for_idle(target, &abstractcs) != ERROR_OK)
 		goto clear_abstractauto_and_fail;
 
-	info->cmderr = get_field(abstractcs, DM_ABSTRACTCS_CMDERR);
-	switch (info->cmderr) {
+	uint32_t cmderr = get_field(abstractcs, DM_ABSTRACTCS_CMDERR);
```

Comment:
In some places, cmderr has type of uint32_t and in others it is unsigned int.

Please, could you unify it.

Also when assigning to uint32_t, consider using get_field32() so that we avoid potential compiler warnings.

Please, take a look at all the similar get_field() cases in this pull request. I haven't commented on each of them individually.

Reply:
> Please, could you unify it.

I think I've  tried to do that, but the final scope of changes was quite large. I'll take a look once again, but **if** the number of changes is large, I would prefer to address it in a subsequent commit.

Reply:
If you could double-check the lines with get_field(), affected by this merge request, that would be nice.

As for the other uses of get_field(), I agree that these can (and should) be handled in a separate pull request.

Reply:
double checked. This was the only place - I'll adjust.